spec/src/main/asciidoc/core/definition.asciidoc

```diff
@@ -81,7 +81,7 @@ The bean types of a bean are used by the rules of typesafe resolution defined in
 
 Almost any Java type may be a bean type of a bean:
 
-* A bean type may be an interface, a concrete class or an abstract class, and may be declared final or have final methods.
+* A bean type may be an interface, a concrete class or an abstract class or a record, may be declared sealed or non-sealed or final, and may have final methods.
```

Comment:
```suggestion
* A bean type may be an interface, a concrete class, an abstract class or a record. It may be declared sealed, non-sealed or final, and may have final methods.
```
Just a minor suggestion. The original sentence is somewhat long and hard to chew on, so splitting it in two might help.

Reply:
I found this confusing. Records and Sealed classes cannot be managed beans. Why do we call them out for Bean Types. Can you list a use case to support such a Record or Sealed Classes as bean types and then inject them to something else. I would not add them here at all.

Reply:
Andrew's comment nicely sums this up https://github.com/jakartaee/cdi/pull/693#issuecomment-1755584620

Generally the use case is similar to why you would want to have a non-proxyable class (such as with no no-arg constructor) to be a bean - it is doable but only works until you try to assign it normal scope or use an interceptor.
We didn't have a specific case in mind; instead, the idea is to avoid introducing extra rules where we can apply what CDI already does.

Reply:
Records and sealed types _can_ be managed beans. (They are unproxyable, so cannot be normal scoped or intercepted or decorated, but it's perfectly legal to make a record `@Dependent`, as long as it defines a zero-param constructor, which it can. It is not very practical, but it is possible.)

But that's not the point here. There are other kinds of types that cannot be managed beans (such as interfaces) yet it makes perfect sense for them to be bean types. Maybe you're confusing bean types and bean classes?

Reply:
The point is that it is VERY problmatically. I recommend not to list them out. If we call it out here, it encourages end users to use Records in anger if they just do a quick search on the doc.
The existing phrase `A bean type may be an interface, a concrete class or an abstract class, and may be declared final or have final methods.` covers to Records, sealed classes.

Reply:
> If we call it out here, it encourages end users to use Records in anger if they just do a quick search on the doc.

That is _not_ an argument for or against anything. This is a specification, not a microwave oven manual.

You could argue that per JLS, a record class is "a restricted kind of class", so whatever applies to classes applies to records as well, which would be correct. So we don't necessarily have to call out records.

But that doesn't cover sealed types. We specify that a bean type may be final, but we don't specify whether it may be sealed, and until we do, it remains unspecified and hence non-portable.

Reply:
okay, just add sealed types then. 
I am also wondering whether to clearly say what kind of injections an end user can do with Records. In order to inject a Record, it must be `@Dependent` scoped and it must define a zero-arg constructor. Thoughts?

Reply:
Records may be made beans by following the existing rules. Records are always final, so they are unproxyable, which means that a record bean cannot be normal scoped. That leaves `@Singleton` and `@Dependent`. It also has to have a zero-param constructor or an `@Inject` constructor. All the usual. If you don't want to mention records here (because users would want to use them and because the information is redundant), why would we want to mention them elsewhere (causing the same "problems")?

Reply:
okay, fine with not mentioning Record.

Reply:
Done.